src/client/jupyter/kernel-manager.ts

```diff
@@ -78,26 +78,25 @@ export class KernelManagerImpl extends EventEmitter {
         });
     }
 
-    public startKernelFor(language: string): Promise<Kernel> {
-        return this.getKernelSpecFor(language).then(kernelSpec => {
-            return this.startKernel(kernelSpec, language);
-        });
+    public async startKernelFor(language: string): Promise<Kernel> {
+        const kernelSpec = await this.getKernelSpecFor(language);
+        return this.startKernel(kernelSpec, language);
     }
 
     public startExistingKernel(language: string, connection, connectionFile): Promise<Kernel> {
         throw new Error('Start Existing Kernel not implemented');
     }
 
-    public startKernel(kernelSpec: KernelspecMetadata, language: string): Promise<Kernel> {
+    public async startKernel(kernelSpec: KernelspecMetadata, language: string): Promise<Kernel> {
```

Comment:
Yes, I need to fix some test issues, looks like the factoring isn't working anymore. Similar issues have Been reported by users too.